dvc/remote/base.py

```diff
@@ -267,8 +275,7 @@ def load_dir_cache(self, checksum):
             with self.cache.open(path_info, "r") as fobj:
                 d = json.load(fobj)
         except (ValueError, FileNotFoundError):
-            logger.exception("Failed to load dir cache '{}'".format(path_info))
-            return []
+            raise DirCacheError(checksum)
```

Comment:
```suggestion
            raise DirCacheError(checksum, cause=exc)
```

Reply:
Please don't forget to chain exceptions. Otherwise, we won't have them when we need them the most.